New sketch 3A3A1 - Flight control stick

Thibaud-54 opened the pull request:
## Description

Here is the flight controller script. It has been tested on the complete assembly, with the mechanical and electronic components compliant to OH specifications. The template for creating the sketch was applied following issue #41. A Readme text file has also been added as an appendix to the script to provide users with a clear understanding of the integrated features.

Closes #41 

### Dependencies
List of libraries required :
 - SPI
 - Joystick by M.Heironimus
 - EEPROM

### Type of change
- [x] New software module (new software module for slave)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update outside of the automatically-generated Doxygen documentation.

### Checklist:
- [x] My code follows the [style guidelines](https://jrsteensen.github.io/OpenHornet-Software/d4/d46/md__2github_2workspace_2_s_t_y_l_e_g_u_i_d_e.html) of this project
- [x] [I have complied with the software manual](https://jrsteensen.github.io/OpenHornet-Software/d7/d78/md__software_manual.html) for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code fully with Doxygen compatible comments, particularly in hard-to-understand areas
- [x] I have made corresponding changes to non-Doxygen generated documentation
- [ ] I have ran Doxygen locally, and it builds the docs successfully
- [x] My changes generate no errors on compile in Arduino IDE
- [x] My changes generate no new warnings on compile in Arduino IDE
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (For sketches only) This sketch complies with OH-INTERCONNECT v**10**
- [x] If this sketch requires additional libraries, [I have added it as a sub-module per the Arduino Libraries section of the Software Manual.](https://jrsteensen.github.io/OpenHornet-Software/d7/d78/md__software_manual.html)

### How Has This Been Tested?

- [x] I have tested the sketch in-circuit in DCS.

#### Description of Testing
Fully tested in both Windows Properties and DCS with a fully assembled OH flight stick. 

#### Test Configuration

jrsteensen committed:
Merge pull request #93 from jrsteensen/develop

Incremental PR from `develop` to `main`.

jrsteensen committed:
Update ci.yaml

Updated depreciated cache version. (Was: cache@v4.0.1; Is: cache@v4

Thibaud-54 committed:
New sketch 3A3A1 - Flight control stick

Here is the flight controller script. It has been tested on the complete assembly, with the mechanical and electronic components compliant to OH specifications. The template for creating the sketch was applied following issue #41. A Readme text file has also been added as an appendix to the script to provide users with a clear understanding of the integrated features.

jrsteensen reviewed (changes requested):
- [ ] Change base branch to be against develop ([howto](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request))
- [ ] Add makefile like other sketches in repo to tell the workflow what libraries to test it against and the hardware used.
- [ ] Add new libraries as submodules.
- [ ] Add doxygen compatible comments. (See /STYLEGUIDE.md and /docs/SoftwareManual.md on the develop branch)

Thibaud-54 committed:
Updated files

base branch changed.
Makefile added
No new libraries to add.
doxygen compatible comments added to ino file.

Thibaud-54 commented:
> * [x]  Change base branch to be against develop ([howto](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request))
> * [x]  Add makefile like other sketches in repo to tell the workflow what libraries to test it against and the hardware used.
> * [ ]  Add new libraries as submodules. **There is no new lib to add (Joystick is already added to lib folder. SPI and EEPROM should not be added I think).**

> * [x]  Add doxygen compatible comments. (See /STYLEGUIDE.md and /docs/SoftwareManual.md on the develop branch) 
**First time I do it, thank you to confirm if it is ok or not.**

Thibaud-54 committed:
Revert "Update ci.yaml"

This reverts commit 14561e52d56b5365281901e3f77142c4f0f79e1c.

jrsteensen reviewed (approved):
At some point, we should move the README entirely into the ino file with doxygen formatting so it appears in the web accessible documentation instead of having people have to search for it. Approving as-is for now. Now to get it through checks.

jrsteensen committed:
Merge branch 'develop' into feature-new_sketch-3A3A1-STICK_CONTROLLER
